MC/config/PWGHF/external/generator/generator_pythia8_gaptriggered_hf.C

```diff
@@ -88,54 +137,60 @@ protected:
     return true;
   }
 
-  bool selectEvent(const Pythia8::Event& event)
+  bool selectEvent(const Pythia8::Event &event)
   {
-    bool isGoodAtPartonLevel;
-    bool isGoodAtHadronLevel;
-    if(mDoNoQuarkTrigger){
-      isGoodAtPartonLevel = (mHadronPdg != 0) ? true : false;
-      isGoodAtHadronLevel = (mHadronPdg != 0) ? false : true;
-    } else {
-      isGoodAtPartonLevel = false;
-      isGoodAtHadronLevel = (mHadronPdg != 0) ? false : true;
-    }
 
-    for (auto iPart{0}; iPart < event.size(); ++iPart) {
+    bool isGoodAtPartonLevel{mQuarkPdgList.size() == 0};
+    bool isGoodAtHadronLevel{mHadronPdgList.size() == 0};
+
+    for (auto iPart{0}; iPart < event.size(); ++iPart)
+    {
 
       // search for Q-Qbar mother with at least one Q in rapidity window
-      if (!isGoodAtPartonLevel) {
+      if (!isGoodAtPartonLevel)
+      {
         auto daughterList = event[iPart].daughterList();
         bool hasQ = false, hasQbar = false, atSelectedY = false;
-        for (auto iDau : daughterList) {
-          if (event[iDau].id() == mQuarkPdg) {
+        for (auto iDau : daughterList)
+        {
+          if (event[iDau].id() == mQuarkPdg)
+          {
             hasQ = true;
-            if ((event[iDau].y() > mQuarkRapidityMin) && (event[iDau].y() < mQuarkRapidityMax)) {
+            if ((event[iDau].y() > mQuarkRapidityMin) && (event[iDau].y() < mQuarkRapidityMax))
+            {
               atSelectedY = true;
             }
           }
-          if (event[iDau].id() == -mQuarkPdg) {
+          if (event[iDau].id() == -mQuarkPdg)
+          {
             hasQbar = true;
-            if ((event[iDau].y() > mQuarkRapidityMin) && (event[iDau].y() < mQuarkRapidityMax)) {
+            if ((event[iDau].y() > mQuarkRapidityMin) && (event[iDau].y() < mQuarkRapidityMax))
+            {
               atSelectedY = true;
             }
-          }  
+          }
         }
-        if (hasQ && hasQbar && atSelectedY) {
+        if (hasQ && hasQbar && atSelectedY)
+        {
           isGoodAtPartonLevel = true;
         }
       }
 
       // search for hadron in rapidity window
-      if (!isGoodAtHadronLevel) {
+      if (!isGoodAtHadronLevel)
+      {
         int id = std::abs(event[iPart].id());
         float rap = event[iPart].y();
-        if (id == mHadronPdg && rap > mHadRapidityMin && rap < mHadRapidityMax) {
+        if (id == mHadronPdg && rap > mHadRapidityMin && rap < mHadRapidityMax)
+        {
           isGoodAtHadronLevel = true;
         }
       }
 
       // we send the trigger
-      if (isGoodAtPartonLevel && isGoodAtHadronLevel) {
+      if (isGoodAtPartonLevel && isGoodAtHadronLevel)
+      {
+        LOG(info)<<"EVENT SELECTED: Found particle "<<event[iPart].id()<<" at rapidity "<<event[iPart].y()<<"\n";
```

Comment:
To avoid many messages I would put `debug` here
```suggestion
        LOG(debug)<<"EVENT SELECTED: Found particle "<<event[iPart].id()<<" at rapidity "<<event[iPart].y()<<"\n";
```